cstring.h

```diff
@@ -895,9 +888,66 @@ void stringRemove(TString *s, size_t pos, size_t len) {
 void stringDestroy(TString *s) {
     if (s == NULL) return;
     free(s->data);
-    *s = (TString) {0};
+    *s = (TString){0};
+}
+
+double stringToDouble(TString s) {
+    size_t number = 0;
+    size_t decimal = 0;
+    size_t p = 1;
+    bool in_decimal = false;
+    char current;
+    if (s.data[0] == '-') {
+        for (int i = 1; i < s.size; i++) {
+            current = s.data[i];
+            if (current == '1' || current == '2' || current == '3' || current == '4' ||
+                current == '5' || current == '6' || current == '7' || current == '8' ||
+                current == '9' || current == '0' || current == ',' || current == '.') {
+                if (strchr(".,", current)) {
+                    in_decimal = true;
+                } else {
+                    if (in_decimal) {
+                        if (p == 1) {
+                            decimal = (int)(current - '0');
+                        } else {
+                            decimal *= 10;
+                            decimal += (int)(current - '0');
+                        }
+                        p *= 10;
+                    } else {
+                        number *= 10;
+                        number += (int)(current - '0');
+                    }
+                }
+            }
+        }
+        return -(number + ((double)(decimal)) / p);
```

Comment:
It may lead to integer overflow when string is long enough. Let's have two float values: integer and fractional parts of number. In the end we should just add them.